pkg/operator/controllers/machinehealthcheck/machinehealthcheck_controller.go

```diff
@@ -32,50 +32,63 @@ var machinehealthcheckYaml []byte
 var mhcremediationalertYaml []byte
 
 const (
-	ControllerName string = "MachineHealthCheck"
-	managed        string = "aro.machinehealthcheck.managed"
-	enabled        string = "aro.machinehealthcheck.enabled"
+	MHCControllerName string = "MachineHealthCheck"
+	MHCManaged        string = "aro.machinehealthcheck.managed"
+	MHCEnabled        string = "aro.machinehealthcheck.enabled"
 )
 
-type Reconciler struct {
-	log *logrus.Entry
+type MachineHealthCheckReconciler struct {
+	base.AROController
 
 	dh dynamichelper.Interface
-
-	client client.Client
 }
 
-func NewReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *Reconciler {
-	return &Reconciler{
-		log:    log,
-		dh:     dh,
-		client: client,
+func NewMachineHealthCheckReconciler(log *logrus.Entry, client client.Client, dh dynamichelper.Interface) *MachineHealthCheckReconciler {
+	return &MachineHealthCheckReconciler{
+		AROController: base.AROController{
+			Log:    log,
+			Client: client,
+			Name:   MHCControllerName,
+		},
+		dh: dh,
 	}
 }
 
-func (r *Reconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	instance := &arov1alpha1.Cluster{}
-	err := r.client.Get(ctx, types.NamespacedName{Name: arov1alpha1.SingletonClusterName}, instance)
+// Reconcile watches MachineHealthCheck objects, and if any changes,
+// reconciles the associated ARO MachineHealthCheck object
+
```

Comment:
I'm not 100% sure, but I think this extra newline might mess up godoc on this function.

Reply:
So should  I just remove the blank lines or the comment also? Because other ones are just comments on it.